handlers/extensions.go

```diff
@@ -19,6 +21,8 @@ type ExtensionVersion struct {
 }
 
 func (h *Handler) ExtensionsEndpointHandler(w http.ResponseWriter, req *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+	mx.Lock()
+	defer mx.Unlock()
```

Comment:
@Revolyssup are we holding for a longer than required? I am not sure but it seems like that we won't release the lock unless the request context dies, right?

Reply:
I did this because the variable is still global. I can use the lock just around where the variable is accessed but if someone else tomorrow edits the code inside this function, maybe they dont keep that in mind so I did this.
If you say so, I will lock just around the line variable is read/write. And I will add the comment above the variable to use it in a thread safe way. Sound right?